scripts/pi-hole/js/network.js

```diff
@@ -108,14 +110,34 @@ $(document).ready(function () {
       // Set hostname to "unknown" if not available
       if (!data.name || data.name.length === 0) {
         $("td:eq(3)", row).html("<em>unknown</em>");
+      } else {
+        var names = [];
+        maxiter = Math.min(data.name.length, MAXIPDISPLAY);
+        index = 0;
```

Comment:
This seems to be set in the for loop, the same for the other for loop bellow.

Reply:
In fact, you probably don't need the `index` variable in line 72. Can you try just doing `for (var index = 0; index < maxiter; index++)` in both loops and remove the upper `index` var and its assignments?

Reply:
This is probably the remainder of several changes over time. I will check what to do best when this PR reaches maturity (I will probably do it before, even).

Reply:
As far as I understand this, JavaScript, unlike the other languages I'm typically using, does not support block level scoping, right? This means that `var index`, even when defined inside of a block structure like this `for` loop, does not restrict that variable to the loop itself. Instead, the variable will be accessible from the entire function and specifying
```javascript
for (var index = 0; index < maxiter; index++)
```
will result in an error (variable already declared), no?

Reply:
It supports block level variables with `let` and `const` which we are not using.

Just use `i`, `j` etc in the for loop itself :)